amethyst_ui/src/pass.rs

```diff
@@ -137,7 +109,37 @@ impl<'a> PassData<'a> for DrawUi {
 }
 
 impl Pass for DrawUi {
-    fn compile(&self, effect: NewEffect) -> Result<Effect> {
+    fn compile(&mut self, effect: NewEffect, world: &mut World) -> Result<Effect> {
+        // Initialize a single unit quad, we'll use this mesh when drawing quads later
+        let data = vec![
+            PosTex {
+                position: [0., 1., 0.],
+                tex_coord: [0., 0.],
+            },
+            PosTex {
+                position: [1., 1., 0.],
+                tex_coord: [1., 0.],
+            },
+            PosTex {
+                position: [1., 0., 0.],
+                tex_coord: [1., 1.],
+            },
+            PosTex {
+                position: [0., 1., 0.],
+                tex_coord: [0., 0.],
+            },
+            PosTex {
+                position: [1., 0., 0.],
+                tex_coord: [1., 1.],
+            },
+            PosTex {
+                position: [0., 0., 0.],
+                tex_coord: [0., 1.],
+            },
+        ].into();
+        let loader: &Loader = &*world.read_resource();
+        let mesh_storage = &*world.read_resource();
+        self.mesh_handle = Some(loader.load_from_data(data, (), mesh_storage));
```

Comment:
You don't need to create quad with loader and store it in the world.
Instead I suggest to use `gfx::Factory` directly. And store mesh inside `DrawUi`.

Reply:
In this scenario how would I obtain the `gfx::Factory`?  Should it also be passed to the compile function similar to the `specs::World`?

Reply:
It is already there inside `NewEffect`

Reply:
@omni-viral Oh!  Thanks!  I actually didn't remember that.

Reply:
Well. It is in line you've changed. With rustfmt I presume :smile: